tests/integration/cloud/end_to_end/conftest.py

```diff
@@ -37,31 +41,119 @@ def context() -> CloudDataContext:
 
 
 @pytest.fixture(scope="module")
-def tmp_dir(tmpdir_factory) -> py.path:
-    return tmpdir_factory.mktemp("project")
+def datasource_name(
+    context: CloudDataContext,
+) -> Iterator[str]:
```

Comment:
```suggestion
) -> Generator[str, None, None]:
```
`#non-blocking`
You could also type this like this.
https://docs.python.org/3/library/typing.html#typing.Generator

Reply:
I find `Generator[str, None, None]` more confusing. The new developer has to spend the time figuring out what the `None`s represent instead of just using it like an `Iterator`.